### PR TITLE
Add `fn_type` for type enums

### PR DESCRIPTION
**Pull request by gavrilikhin-d**
## Description

Adds implementation of `fn_type` for type enums, for convenience

## Related Issue

#380 

## How This Has Been Tested

Run doc tests and `cargo test --features llvm14-0`

## Checklist

- [X] I have read the [Contributing Guide](https://github.com/TheDan64/inkwell/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: src/types/enums.rs [hunk 1]
```diff
@@ -104,6 +104,66 @@ enum_type_set! {
...
+    pub fn fn_type(self, param_types: &[BasicMetadataTypeEnum<'ctx>], is_var_args: bool) -> FunctionType<'ctx> {
+        use BasicMetadataTypeEnum::*;
+        match self {
+            ArrayType(ty) => ty.fn_type(param_types, is_var_args),
+            FloatType(ty) => ty.fn_type(param_types, is_var_args),
+            IntType(ty) => ty.fn_type(param_types, is_var_args),
+            PointerType(ty) => ty.fn_type(param_types, is_var_args),
+            StructType(ty) => ty.fn_type(param_types, is_var_args),
+            VectorType(ty) => ty.fn_type(param_types, is_var_args),
+            // TODO: are functions really able to return metadata?
```
Comment (TheDan64) [inline src/types/enums.rs:129]: We should verify this TODO before adding this function. If it's not valid, it should return a Result

Comment (gavrilikhin-d) [reply src/types/enums.rs:129]: Ok, I'll check it

Comment (gavrilikhin-d) [reply src/types/enums.rs:129]: I tried verification of module with function has metadata as return type. It works. But I never found anything about returning metadata on official documentation.

Comment (gavrilikhin-d) [reply src/types/enums.rs:129]: Any ideas what to do?

Comment (TheDan64) [reply src/types/enums.rs:129]: If it passed verification then I suppose that's fine

**Comment by TheDan64** [comment]
This needs to be rebased and might need some slight tweaks to support LLVM 15? Not certain

**Pull request closed without merging**